### ISSUE TITLE
Misunderstood pydantic PrivateAttr initialization in the constructor

### ISSUE THREAD

**Issue by irgolic** [microsoft/pyright#7209]
**Describe the bug**
Pydantic models with private attributes are incorrectly assumed to be valid in a constructor, and `__init__` overrides of parent classes are ignored.

**Code or Screenshots**

Passing a PrivateAttr's value to a pydantic constructor does not set its value.

```python
import pydantic


class A(pydantic.BaseModel):
    _cache: bool = pydantic.PrivateAttr(default=True)


# the following should flag a type error but it doesn't
a = A(_cache=False)
print(a._cache)  # True
```

The `__init__` signature of the parent class is forgotten by pyright in the subclass:

```python
import pydantic


class A(pydantic.BaseModel):
    _cache: bool = pydantic.PrivateAttr(default=True)

    def __init__(self, cache: bool = True, **kwargs):
        super().__init__(**kwargs)
        self._cache = cache

# as expected
a = A(cache=False)
print(a._cache)  # False


class B(A):
    pass


# the following line incorrectly flags a type error
b = B(cache=False)  # error: No parameter named "cache" (reportCallIssue)
print(b._cache)  # False
```

At the moment, this is the pyright-compatible way of using PrivateAttr:
```python
import pydantic


class A(pydantic.BaseModel):
    _cache: bool = pydantic.PrivateAttr(default=True)

    def __init__(self, _cache: bool = True, **kwargs):
        super().__init__(**kwargs)
        self._cache = _cache


a = A(_cache=False)
print(a._cache)  # False


class B(A):
    pass


b = B(_cache=False)
print(b._cache)  # False
```

Versions:
```
pydantic==2.5.3
pydantic_core==2.14.6
```

**VS Code extension or command-line**
Pyright 1.1.350 CLI

**Comment by samuelcolvin** [comment]
I agree with @erictraut. (I might get a t-shirt with that on it)

---

If there's something dirty but inconsequential we can do to fool pyright and others into not raising errors here, happy to try.

Otherwise the only solution is to get python itself to accept some more typing concepts, until then `# type ignore` is your friend.

**Comment by irgolic** [comment]
Thanks for both your thoughts and quick replies.

Personally I will employ this pattern in my code:

```python
class A(pydantic.BaseModel):
    _cache: bool = pydantic.PrivateAttr(default=True)

    def __init__(self, _cache: bool = True, **kwargs):
        super().__init__(**kwargs)
        self._cache = _cache
```

Which seems to bring it back in line with PEP 681.

Do you know the original reason for not setting `PrivateAttr`s via constructor @samuelcolvin?
